src/tree/registries/RegistriesTreeItem.ts

```diff
@@ -140,13 +140,35 @@ export class RegistriesTreeItem extends AzExtParentTreeItem {
         await this.saveCachedProviders();
     }
 
-    public async disconnectRegistry(context: IActionContext, node: IRegistryProviderTreeItem): Promise<void> {
-        context.telemetry.properties.providerId = node.cachedProvider.id;
-        context.telemetry.properties.providerApi = node.cachedProvider.api;
+    public async disconnectRegistry(context: IActionContext, cachedProvider: ICachedRegistryProvider | undefined): Promise<void> {
+        if (!cachedProvider) {
+            const picks = this._cachedProviders.map(crp => {
+                const provider = getRegistryProviders().find(rp => rp.id === crp.id);
+                let label: string = (provider && provider.label) || crp.id;
+                let descriptions: string[] = [];
+                if (crp.username) {
+                    descriptions.push(`Username: "${crp.username}"`)
+                }
+                if (crp.url) {
+                    descriptions.push(`URL: "${crp.url}"`)
```

Comment:
Why are you pushing these strings to an array and then invoking them by index, instead of just assigning these strings to a `description` and a `detail` variable? It also seems buggy - if `crp` doesn't have a url or username field, an error would be thrown when assigning a value to `returnValue.detail`.

Reply:
> It also seems buggy - if crp doesn't have a url or username field, an error would be thrown when assigning a value to returnValue.detail.

I don't think arrays are as strict as you think they are. You can do `descriptions[9999]` with an empty array and it'll return `undefined` with no error.

> Why are you pushing these strings to an array and then invoking them by index, instead of just assigning these strings to a description and a detail variable?

It's not a simple assignment because `url` and/or `username` could be undefined and I never want to be in a case where `detail` is defined, but `description` isn't (since `description` takes priority). I probably could've more like what you're asking, but it feels like one and the same.